tools/ci_build/github/azure-pipelines/templates/android-java-api-aar-test.yml

```
@@ -59,7 +64,20 @@ jobs:
     workingDirectory: $(Build.BinariesDirectory)
 
   # skip emulator tests for qnn package as there are no arm64-v8a emulators and no qnn libraries for x86
-  - ${{ if not(contains(parameters.packageName, 'qnn')) }}:
+  # skip emulator tests for webgpu package as software rendering with swiftshader does not work without host GPU
+  - script: |
+        set -e
+        if [[ "${{ parameters.packageName }}" == *"qnn"*  || "${{ parameters.enableWebGpu }}" == "true" ]]; then
```

can we also include a comment about it so that it is obvious for other devs who may modify this later?
```suggestion
        # Do not output ##vso[] commands with `set -x` or they may be parsed again and include a trailing quote. 
        if [[ "${{ parameters.packageName }}" == *"qnn"*  || "${{ parameters.enableWebGpu }}" == "true" ]]; then
```